### PR TITLE
sessionctx: Error/warning on unsupported isolation levels

### PR DESCRIPTION
**Pull request by morgo**
### What problem does this PR solve? 

Fixes https://github.com/pingcap/tidb/issues/2712

### What is changed and how it works?

For isolation levels, the following behavior is provided:
- read-uncommitted: error (unsupported)
- read-committed:  clean
- repeatable-read: clean
- serializable: error (unsupported)

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

Code changes

 - minimal

Side effects

 - Breaking backward compatibility (there may be some apps which depend on `read-uncommitted` or `serializable`, but they are less common than RR and RC)

Related changes

 - Need to update the documentation
 - Need to be included in the release note

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8625)

### CODE REVIEW

File: sessionctx/variable/varsutil.go [hunk 1]
```diff
@@ -394,6 +394,15 @@ func ValidateSetSystemVar(vars *SessionVars, name string, value string) (string,
 		if !exists {
 			return "", ErrWrongValueForVar.GenWithStackByArgs(name, value)
 		}
+		switch upVal {
+		case "SERIALIZABLE":
+			return "", ErrUnsupportedValueForVar.GenWithStackByArgs(name, value)
+		case "READ-COMMITTED", "READ-UNCOMMITTED":
+			// prevent double warning
+			if name == "transaction_isolation" {
+				vars.StmtCtx.AppendWarning(ErrUnsupportedValueForVar.GenWithStackByArgs(name, value))
```
Comment (winkyao) [inline sessionctx/variable/varsutil.go]: Why not return ErrUnRecommented?

Comment (winkyao) [reply sessionctx/variable/varsutil.go]: Well, I saw your comment, but I think we can create a new class in this PR.

**Comment by morgo** [comment]
I have changed `read-uncommitted` to be an error now, and updated the description.  `read-committed` also uses a new error class of `NotRecommended`.

**Comment by morgo** [comment]
It looks like this does break the JDBC test:
```
Loading JDBC driver 'com.mysql.jdbc.Driver'

Connected to 5.7.10-TiDB-v2.1.0-rc.3-299-g0dc57f7

[ERROR] Tests run: 30, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 4.957 s <<< FAILURE! - in pingcap.tidb.test.simple.ConnectionTest

[ERROR] testIsolationLevel(pingcap.tidb.test.simple.ConnectionTest)  Time elapsed: 0.01 s  <<< ERROR!

java.sql.SQLException: variable 'tx_isolation' does not yet support value: READ-UNCOMMITTED

	at pingcap.tidb.test.simple.ConnectionTest.testIsolationLevel(ConnectionTest.java:553)
```

This is the side-effect mentioned in the PR.  Some applications will break on warnings too, particularly since `read-committed` is frequently used.

**Comment by morgo** [comment]
I was concerned that a warning for `read-committed` may break too many applications, so I've changed it to be clean again.  The description has been updated.

**Comment by disksing** [comment]
LGTM.

**Pull request merged**